_posts/2024-08-07-taming-apache-kafka-4.0.md

```diff
@@ -0,0 +1,187 @@
+
+#### ZooKeeper support
+
+As already explained in one of the previous sections, the current expectation is that the last Apache Kafka release with Apache ZooKeeper support will be Kafka 3.9.
+Once it is released, Strimzi will release a new version with support for Apache Kafka 3.8 and 3.9.
+Depending on the exact timing, there might be two such Strimzi versions:
+* One version (let's call it `0.4x`) soon after the Kafka 3.9 release
+* Another one (let's call it `0.4y`) approximately 1-2 months later
+
+Kafka 3.9 is expected to address at least some of the remaining KRaft issues.
+In particular, it should add support for controller scaling.
+So in these Strimzi versions, we will try to add support for the new KRaft features.
+The `0.4y` will be the last version that will allow you to run ZooKeeper-based Kafka clusters and let you migrate to KRaft.
+
+The next Strimzi version (let's call it `0.4z`) that will follow will already focus on Kafka 4.0 and will completely remove the ZooKeeper support.
+It will still keep the support for Kafka 3.9 to make the upgrades easier.
+But even with Kafka 3.9, it will let you use KRaft-based Kafka clusters only.
+That means that to upgrade to Strimzi `0.4z`, you will need to be already using KRaft.
+
+The ZooKeeper to KRaft migration has been already supported by Strimzi for some time.
+So while you do not have to migrate right now, you can give it a try, get an idea of how it works, and help us test it today.
+You can read more about it in our [blog post about KRaft migration](https://strimzi.io/blog/2024/03/22/strimzi-kraft-migration/).
+
+If you don't migrate your Kafka cluster to KRaft before upgrading to Strimzi `0.4z`, Strimzi will not delete or damage the Kafka cluster.
+But it will not be able to operate such a cluster.
+So you will have to either delete the cluster and create a new one based on KRaft.
+Or downgrade Strimzi and migrate the cluster to KRaft.
+
+The following table provides an overview of the different versions and their support for ZooKeeper and KRaft:
+
+| Version | Supported Kafka versions | Supported metadata modes |
+| :-----: |:-------------------------|:-------------------------|
+| ...     | 3.7.x, 3.8.x             | ZooKeeper, KRaft         |
+| 0.4x    | 3.8.x, 3.9.x             | ZooKeeper, KRaft         |
+| 0.4y    | 3.8.x, 3.9.x             | ZooKeeper, KRaft         |
+| 0.4z    | 3.9.x, 4.0.x             | KRaft                    |
+| ...     | 4.0.x, 4.1.x             | KRaft                    |
+
+For more details and other alternatives we considered, please check out the [_Support for Apache Kafka 4.0_](https://github.com/strimzi/proposals/blob/main/077-support-for-kafka-4.0.md) proposal.
+
+#### Mirror Maker 1 removal
+
+To make things easier to remember, we decided to remove the support for Mirror Maker 1 using the same schedule as we have for the ZooKeeper support removal.
+So the `0.4y` version will be the last Strimzi version with support for Mirror Maker 1.
```

Comment:
```suggestion
So the `0.4y` version will be the last Strimzi version with support for MirrorMaker 1.
```

Reply:
If we want to use the standard naming convention, we should change to _MirrorMaker_ throughout the post

Reply:
I updated it to `MirrorMaker` everywhere.